core/src/main/java/tc/oc/pgm/filters/FilterParser.java

```diff
@@ -443,4 +442,52 @@ public StructuralLoadFilter parseStructuralLoad(Element el) throws InvalidXMLExc
   public TimeFilter parseTimeFilter(Element el) throws InvalidXMLException {
     return new TimeFilter(XMLUtils.parseDuration(el, null));
   }
+
+  @MethodParser("match-phase")
+  public MatchPhaseFilter parseMatchPhase(Element el) throws InvalidXMLException {
+    return parseMatchPhaseFilter(el.getValue(), el);
+  }
+
+  @MethodParser("match-started")
+  public MatchPhaseFilter parseMatchStarted(Element el) throws InvalidXMLException {
+    return parseMatchPhaseFilter("started", el);
+  }
+
+  @MethodParser("match-running")
+  public MatchPhaseFilter parseMatchRunning(Element el) throws InvalidXMLException {
+    return parseMatchPhaseFilter("running", el);
+  }
+
+  @MethodParser("match-finished")
+  public MatchPhaseFilter parseMatchFinished(Element el) throws InvalidXMLException {
+    return parseMatchPhaseFilter("finished", el);
+  }
+
+  private MatchPhaseFilter parseMatchPhaseFilter(String matchState, Element el)
```

Comment:
I have an idea to make this easier:

* Change the signature of `parseMatchPhaseFilter` return a generic `Filter`
* Change `MatchPhaseFilter` to only accept 1 `MatchPhase` and not a `Set`
* For the "started" case, return an `OrFilter(MatchPhaseFilter(RUNNING), MatchPhaseFilter(FINISHED))`

See what I'm getting at?

Reply:
Yes! Great idea 👍